### PR TITLE
Adds departmental banners

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: You can now use metal rods and departmental jumpsuits to craft departments for each banner.
/:cl:

Ever wanted to make nations even more tribalistic, or give a seceding Cargo some flair? Now you can. Banners have been sprited and added for each department, including Securistan, Meditopia, Sciencia, Cargonia, Engitopia, and Command. You can craft them with two metal rods and a jumpsuit related to each department; they are purely cosmetic and have no use on their own.

There are also admin-only banners that can be used to inspire crew around you. Like old HoG banners, these banners can be picked up and waved around to inspire members of the same department, reducing stuns and healing brute and burn damage. There's a one-minute cooldown between uses. Additionally, some banners have additional effects and conditions:

* The Meditopia banner inspires the unconscious as well as Medical staff, and heals toxin/oxygen damage in addition to stabilizing those in critical condition.
* The Engitopia banner removes radiation.
* The Sciencia banner inspires those on fire as well as Science staff.
* The Command banner inspires mindshielded crew as well as Command staff.

They're also a respectable blunt weapon, if you're into that sort of thing.

Finally, optimized banner code a little bit, and gave them framework for doing more neat things.

### CODE REVIEW

**Comment by ike709** [comment]
Hail Cargonia!

**Comment by Bawhoppen** [comment]
Add normal ones that don't do any special effects?

**Comment by Xhuis** [comment]
# From the PR body:
>I'm planning to implement these in some form, gameplay-wise, but I as of yet am not sure how. I might 
add crafting recipes for them, but they would likely lack the inspiration ability due to mass-producable free AoE healing being incredibly overpowered.

**Comment by Bawhoppen** [comment]
I couldn't see why giving normal crew anything that gives arbitrary buffs like this would be good

File: code/game/objects/items/religion.dm [hunk 1]
```diff
@@ -1,27 +1,126 @@
...
-		H.adjustBruteLoss(-15)
-		H.adjustFireLoss(-15)
-		H.AdjustStun(-40)
-		H.AdjustKnockdown(-40)
-		H.AdjustUnconscious(-40)
+	user.visible_message("<span class='big notice'>[user] flourishes [src]!</span>", \
+	"<span class='notice'>You raise [src] skywards, inspiring your allies!</span>")
+	playsound(user, "rustle", 100, FALSE)
+	if(warcry)
+		user.say("[warcry]!!")
```
Comment (RemieRichards) [inline code/game/objects/items/religion.dm]: No, just put the ```!!``` in the warcry if you want them, the way it is now you can't have them say anything that's not a yell.

File: code/game/objects/items/religion.dm [hunk 2]
```diff
@@ -1,27 +1,126 @@
...
+	var/morale_time = 0
+	var/morale_cooldown = 600 //How many deciseconds between uses
+	var/list/job_loyalties //Mobs with any of these assigned roles will be inspired
+	var/list/role_loyalties //Mobs with any of these special roles will be inspired
+	var/warcry
 
 /obj/item/banner/attack_self(mob/living/carbon/human/user)
-	if(moralecooldown + moralewait > world.time)
+	if(morale_time > world.time)
+		to_chat(user, "<span class='warning'>You aren't feeling inspired enough to flourish [src] again yet.")
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: unclosed span

File: code/game/objects/items/religion.dm [hunk 3]
```diff
@@ -1,27 +1,126 @@
...
-	for(var/mob/living/carbon/human/H in range(4,get_turf(src)))
-		to_chat(H, "<span class='notice'>Your morale is increased by [user]'s banner!</span>")
-		H.adjustBruteLoss(-15)
-		H.adjustFireLoss(-15)
-		H.AdjustStun(-40)
-		H.AdjustKnockdown(-40)
-		H.AdjustUnconscious(-40)
+	user.visible_message("<span class='big notice'>[user] flourishes [src]!</span>", \
+	"<span class='notice'>You raise [src] skywards, inspiring your allies!</span>")
+	playsound(user, "rustle", 100, FALSE)
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: play it on src

File: code/game/objects/items/religion.dm [hunk 4]
```diff
@@ -1,27 +1,126 @@
...
+	if(warcry)
+		user.say("[warcry]!!")
+	user.transform *= 1.2
+	animate(user, transform = initial(user.transform), time = 10)
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: Make a copy of the transform before you change it and animate to that instead of initial()

File: code/game/objects/items/religion.dm [hunk 5]
```diff
@@ -1,27 +1,126 @@
...
+		if(H.stat == DEAD)
+			continue
+		if(H.mind && (has_job_loyalties || has_role_loyalties))
+			if(has_job_loyalties && H.mind.assigned_role in job_loyalties)
+				inspired += H
+			else if(has_role_loyalties && H.mind.special_role in role_loyalties)
+				inspired += H
+		else if(check_inspiration(H))
+			inspired += H
+	inspired |= user //The user is always inspired regardless of loyalty
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: Add the user first and then do

```
if(H == user)
    continue
```
in the loop

File: code/game/objects/items/religion.dm [hunk 6]
```diff
@@ -1,27 +1,126 @@
...
+
+	for(var/V in inspired)
+		var/mob/living/carbon/human/H = V
+		if(H != user)
+			to_chat(H, "<span class='notice'>Your confidence surges as [user] flourishes their [name]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: their [src]

**Review by Cyberboss — APPROVED** [review]

**Review by Cyberboss — DISMISSED** [review]
I'm too high rn

File: code/game/objects/items/religion.dm (hunk 1 above)
Comment (Armhulen) [reply code/game/objects/items/religion.dm]: It's a warCRY.

Comment (RemieRichards) [reply code/game/objects/items/religion.dm]: but it doesn't have to be, xhuis already went to the supposed effort of making the system more generic, and then goes and adds something not-generic-at-all.

File: code/game/objects/items/religion.dm [hunk 7]
```diff
@@ -1,27 +1,126 @@
...
+	warcry = "All hail lord Singuloth"
+
+/obj/item/banner/engineering/special_inspiration(mob/living/carbon/human/H)
+	H.radiation = 0
+
+/obj/item/banner/command
+	name = "command banner"
+	desc = "The banner of Command, a staunch and ancient line of bueraucratic kings and queens."
+	//No icon state here since the default one is the NT banner
+	job_loyalties = list("Captain", "Head of Personnel", "Chief Engineer", "Head of Security", "Research Director", "Chief Medical Officer", "Quartermaster")
```
Comment (DoctoreMad) [inline code/game/objects/items/religion.dm]: But Quartermasters aren't...

Comment (Cyberboss) [reply code/game/objects/items/religion.dm]: REEEEEE

**Comment by p440** [comment]
atmosians wont be happy about this

**Comment by Grazyn** [comment]
Rally the banners, we march at dawn.

File: code/game/objects/items/religion.dm (hunk 1 above)
Comment (Supermichael777) [reply code/game/objects/items/religion.dm]: if your waving a banner around chanting LIZARDS WILL NOT REPLACE US! then you already retarded enough to yell all the time

File: code/game/objects/items/religion.dm (hunk 7 above)
Comment (Xhuis) [reply code/game/objects/items/religion.dm]: REEEEEE

File: code/game/objects/items/religion.dm (hunk 6 above)
Comment (Xhuis) [reply code/game/objects/items/religion.dm]: That would lead to "their the meditopia banner"

File: code/game/objects/items/religion.dm (hunk 7 above)
Comment (RemieRichards) [reply code/game/objects/items/religion.dm]: This will need removing.

**Review by RemieRichards — DISMISSED** [review]

**Comment by Xhuis** [comment]
The QM being or not being a head seems subjective; I consider them a head so I included them in the command banner. I think it's fine as-is

**Comment by RemieRichards** [comment]
Your consideration is irrelevant, they AREN'T a head of staff.

**Comment by Xhuis** [comment]
That is your opinion, and I suspect that you'll enforce it if I don't bend to it, so I guess I'm left with no choice.

**Comment by RemieRichards** [comment]
It is not my opinion, they aren't a head of staff, this is fact.
And yes, I will enforce this.

**Comment by Iamgoofball** [comment]
what next, are you going to file a revert over the unnamed gender neutral statue to honor dead coders?

File: code/game/objects/items/religion.dm [hunk 8]
```diff
@@ -1,27 +1,148 @@
...
+	desc = "The banner of the eternal Cargonia, with the mystical power of conjuring any object into existence."
+	icon_state = "banner_cargo"
+	job_loyalties = list("Cargo Technician", "Shaft Miner", "Quartermaster")
+	warcry = "Hail Cargonia!"
+
+/obj/item/banner/cargo/mundane
+	inspiration_available = FALSE
+
+/obj/item/banner/engineering
+	name = "engitopia banner"
```
Comment (PraiseRatvar) [inline code/game/objects/items/religion.dm]: atmosia, not engitopia!!

Comment (Xhuis) [reply code/game/objects/items/religion.dm]: Atmosia and Enginistan formed a peace treaty and united as Engitopia. The Atmosia Declaration of Independence poster you can find proves that its independence is part of history now.

**Comment by kevinz000** [comment]
this is xhuis's way of saying we need to run nations.

**Comment by Bawhoppen** [comment]
I kind of actually agree with Goof

**Comment by ExcessiveUseOfCobblestone** [comment]
The QM is but a usurper

**Comment by Iamgoofball** [comment]
@Xhuis Re-add the cargo banner, Remie having a hissy fit over cosmetic stuff isn't new, despite her turning a blind eye to all furry and anime garbage added.

**Comment by RemieRichards** [comment]
@Iamgoofball who shat in your cereal?

**Comment by RemieRichards** [comment]
Wow I didn't even see the rest, why are you so triggered?
is it cos you lost the election you had no right running in and absolutely zero chance to win because everyone sees you as the joke you actually are?

**Comment by Xhuis** [comment]
Please take it to PMs or somewhere besides this PR. I seriously don't mind removing it. I knew going in that the QM being a head of staff was a touchy subject, so I didn't expect it to go through without it being pointed out anyway.

**Comment by RemieRichards** [comment]
there is no subject about 'the QM being a head' it's never even been a debate, they're not a head.
plain and simple.

**Comment by Xhuis** [comment]
They're a head in our hearts.

**Comment by DoctoreMad** [comment]
@RemieRichards  Lets be honest here, who cares? Being healed by the banners is just a fun thing, and doesn't necessarily have to be built on the standards you've made for them. Alone. In your head.

File: code/game/objects/items/religion.dm (hunk 6 above)
Comment (optimumtact) [reply code/game/objects/items/religion.dm]: @ChangelingRain added a bunch of pronoun procs, please use them https://github.com/tgstation/tgstation/blob/551ede6d35bc955fa83979049441321f46ab9653/code/__HELPERS/pronouns.dm

**Comment by optimumtact** [comment]
add back the cargonia banner, cargo is a department and deserves it's own nations banner, irrespective of if the QM is a head of staff or not.

Theoretically the HoP should take that role, for all of service

**Comment by optimumtact** [comment]
@KorPhaeron do you agree with that?

**Comment by lzimann** [comment]
pointless bloat

**Comment by KorPhaeron** [comment]
QM may not be a head of staff but every single time I have seen nations played "Cargonia" has been a thing

Hell "Cargonia" is a thing outside of nations rounds, it'd be weird to add flags but skip out on the most popular nation.

**Comment by Xhuis** [comment]
I think there's some confusion here; the Cargonia banner is still there. I just removed the Command banner counting the QM as a valid target for inspiration (a joke about how they were a head of staff.)

**Comment by optimumtact** [comment]
fucking goof

**Comment by Xhuis** [comment]
¯\\\_(ツ)_/¯

**Comment by KorPhaeron** [comment]
Oh okay, I don't care about that

**Comment by Xhuis** [comment]
Given that these might realistically never see use as an admin-only thing, I added crafting recipes for the cosmetic (read: can't use to inspire!) variants, so now you can declare independence or just spice up your department with the spare jumpsuits lying around.

They can be crafted with two metal rods and a departmentally-related jumpsuit, including:
* Security officer's jumpsuit
* Medical doctor's jumpsuit
* Scientist's jumpsuit
* Cargo technician jumpsuit
* Engineer's jumpsuit
* Captain's parade jumpsuit

For their respective departments.

**Comment by RemieRichards** [comment]
See, you people choose to listen to fuckhead over there and this is what happens, you get incorrect information and act on it stupidly.

**Comment by DoctoreMad** [comment]
>Most popular nation

If we're going by popularity where is the greytide?

**Comment by Xhuis** [comment]
Unseen and unheard. The Greytide are the silent watchers.

**Comment by DoctoreMad** [comment]
Subtle as 10 assistants going into security and breaking every window you mean

**Comment by Armhulen** [comment]
admin only for now please giving every department cheap infinite heals is silly

**Comment by Xhuis** [comment]
The craftable ones can't be used to heal.

**Comment by Bawhoppen** [comment]
@DoctoreMad Proper term is the GreyTribe

**Comment by DoctoreMad** [comment]
@Bawhoppen Boring.

**Comment by Bawhoppen** [comment]
>boring

**Comment by Iamgoofball** [comment]
Don't believe her lies 
QM is a head
Pluto is a planet

**Comment by DoctoreMad** [comment]
Yes, boring. What, you think you can just rename greytide to greytribe?

**Comment by Bawhoppen** [comment]
@DoctoreMad I never renamed it? People have always used that term on nations rounds. The whole point of this PR is to make reference to such rounds, and in them, players deviate from the normal term (and gameplay) in order to fit the theme of the round, which is supposed to be that of a Medieval to Modern European style diplomacy, conquest, trade, etc. (Hence why departments have new joke names like Cargonia or Atmosia.)
It makes even more sense once you consider the fact that assistants are stereo-typically very adept at combat, and are a large unorganized unruly mass, which fits perfectly in line with the perceived idea of the tribal warrior nomads of the aforementioned time period. 

In short it's a clever reference and pun.
But really, thanks for having to make me explain the entire joke of why nations rounds are funny.

File: code/game/objects/items/religion.dm [hunk 9]
```diff
@@ -8,30 +8,35 @@
 	attack_verb = list("forcefully inspired", "violently encouraged", "relentlessly galvanized")
 	lefthand_file = 'icons/mob/inhands/equipment/banners_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/banners_righthand.dmi'
+	var/inspiration_available = TRUE //If this banner can be used to inspire crew
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: name/desc/examine/something should indicate to players whether or not this is set

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/religion.dm [hunk 10]
```diff
@@ -1,27 +1,196 @@
...
 	icon_state = "banner"
 	item_state = "banner"
+	force = 8
+	attack_verb = list("forcefully inspired", "violently encouraged", "relentlessly galvanized")
 	lefthand_file = 'icons/mob/inhands/equipment/banners_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/banners_righthand.dmi'
-	desc = "A banner with Nanotrasen's logo on it."
-	var/moralecooldown = 0
-	var/moralewait = 600
+	var/inspiration_available = TRUE //If this banner can be used to inspire crew
```
Comment (Cyberboss) [inline code/game/objects/items/religion.dm]: name/desc/examine/something should indicate to players whether or not this is set

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by Xhuis** [comment]
Cargonia lives. Hail Cargonia.
